GitUI/UserControls/SortDiffListContextMenuItem.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.Properties;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public class SortDiffListContextMenuItem : ToolStripMenuItem
+    {
+        private readonly TranslationString _sortByText = new TranslationString("&Sort by...");
+        private readonly TranslationString _filePathSortText = new TranslationString("File &Path");
+        private readonly TranslationString _fileExtensionSortText = new TranslationString("File &Extension");
+        private readonly TranslationString _fileStatusSortText = new TranslationString("File &Status");
+        private readonly IDiffListSortService _sortService;
+        private readonly ToolStripMenuItem _filePathSortItem;
+        private readonly ToolStripMenuItem _fileExtensionSortItem;
+        private readonly ToolStripMenuItem _fileStatusSortItem;
+        private readonly ToolStripMenuItem[] _allItems;
+
+        public SortDiffListContextMenuItem(IDiffListSortService sortService)
+        {
+            _sortService = sortService ?? throw new ArgumentNullException(nameof(sortService));
+            Image = Images.SortBy;
+            Text = _sortByText.Text;
+
+            _filePathSortItem = new ToolStripMenuItem()
+            {
+                Text = _filePathSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FilePath
+            };
+
+            _fileExtensionSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileExtensionSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileExtension
+            };
+
+            _fileStatusSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileStatusSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileStatus
+            };
+
+            _allItems = new[] { _filePathSortItem, _fileExtensionSortItem, _fileStatusSortItem, };
+
+            foreach (var item in AllItems())
+            {
+                item.Click += Item_Click;
+                DropDownItems.Add(item);
+            }
+
+            DropDownOpening += (s, e) => RequerySortingMethod();
+            RequerySortingMethod();
+        }
+
+        private ToolStripMenuItem[] AllItems()
+        {
+            return _allItems;
+        }
+
+        public void RequerySortingMethod()
```

Comment:
This method doesn't need to be `public`. 

If I understand correctly it is make public to be available in tests.
We use a `TestAccessor` pattern to expose/test non public members, e.g.
https://github.com/gitextensions/gitextensions/blob/d4e2919f54232ddc231a88465266f6a337321bdb/GitUI/CommandsDialogs/ToolStripPushButton.cs#L83-L96

Reply:
Will do.